### PR TITLE
Missed migration, fix #216

### LINKED ISSUE #216: Error running "python manage.py migrate"
Hi,

I try to install Sal on Ubuntu 16.04, when I run `python manage.py migrate` I get the following error:

```(sal_env) $ sudo python manage.py migrate
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 305, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 353, in execute
    self.check()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 385, in check
    include_deployment_checks=include_deployment_checks,
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/commands/migrate.py", line 62, in _run_checks
    issues.extend(super(Command, self)._run_checks(**kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/base.py", line 372, in _run_checks
    return checks.run_checks(**kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/core/checks/registry.py", line 81, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/usr/local/lib/python2.7/dist-packages/django/core/checks/urls.py", line 14, in check_url_config
    return check_resolver(resolver)
  File "/usr/local/lib/python2.7/dist-packages/django/core/checks/urls.py", line 24, in check_resolver
    for pattern in resolver.url_patterns:
  File "/usr/local/lib/python2.7/dist-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/usr/local/lib/python2.7/dist-packages/django/urls/resolvers.py", line 310, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/usr/local/lib/python2.7/dist-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/usr/local/lib/python2.7/dist-packages/django/urls/resolvers.py", line 303, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/local/sal_install/sal/sal/urls.py", line 26, in <module>
    url(r'^inventory/', include('inventory.urls')),
  File "/usr/local/lib/python2.7/dist-packages/django/conf/urls/__init__.py", line 50, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/local/sal_install/sal/inventory/urls.py", line 2, in <module>
    from inventory import views
  File "/usr/local/sal_install/sal/inventory/views.py", line 178, in <module>
    class InventoryList(Datatable):
  File "/usr/local/sal_install/sal/inventory/views.py", line 189, in InventoryList
    class Meta:
  File "/usr/local/sal_install/sal/inventory/views.py", line 199, in Meta
    page_length = utils.get_setting('datatable_page_length')
  File "/usr/local/sal_install/sal/server/utils.py", line 323, in get_setting
    setting = SalSetting.objects.get(name=name)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 379, in get
    num = len(clone)
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 238, in __len__
    self._fetch_all()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 1085, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 54, in __iter__
    results = compiler.execute_sql()
  File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 835, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python2.7/dist-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/usr/local/lib/python2.7/dist-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: relation "server_salsetting" does not exist
LINE 1: ...setting"."name", "server_salsetting"."value" FROM "server_sa..
```

### PR DESCRIPTION
**Pull request by grahamgilbert**

### CODE REVIEW

**Comment by clburlison** [comment]
This does fix #216 but now I'm getting an endless loop on the "Send usage data?" popup.

the Server is in Debug mode but nothing gets written out to the logs.

send_usage_data

File: server/utils.py [hunk 1]
```diff
@@ -30,6 +31,10 @@
 TWENTY_FOUR_HOURS = 86400
 
 
+def db_table_exists(table_name):
+    return table_name in connection.introspection.table_names()
```
Comment (clburlison) [inline server/utils.py]: This is the issue. This is always resolving as `False` which allows it to pass the test during migrations.

File: server/utils.py [hunk 2]
```diff
@@ -319,6 +324,10 @@ def get_setting(name, default=None):
         str: Anything else.
 
     """
+
+    # Make sure migrations have run, otherwise return default
+    if not db_table_exists('server_sal_setting'):
```
Comment (clburlison) [inline server/utils.py]: Actually this is the real problem the table name is `server_salsetting`. Swapping that over is resolving the endless loop error.

**Comment by grahamgilbert** [comment]
PTAL @clburlison - now return the actual default values to allow tests to run correctly, and test for the right table name. Next up is to adjust the tests to actually run the migrations to make sure they work.

**Pull request merged**